### DISCUSSION TITLE
cn/2024/09/09/create-py-module/

### DISCUSSION THREAD

**Discussion by giscus** [hongtaoh/hongtaoh.github.io#178] [Announcements]
# cn/2024/09/09/create-py-module/

https://hongtaoh.com/cn/2024/09/09/create-py-module/

**Comment by CyrusYip** [comment]
说点题外话。我发现你好几个 Git 仓库里的提交信息都是「update on 某日期时间」，这太没信息量了，别人无法从中看到项目的大概变化。

推荐你试试 [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)。

**Reply by hongtaoh** [reply]
好问题。我就是太懒了。我在想用 llm 是否可以解决这个问题。

**Reply by CyrusYip** [reply]
我猜应该有这样的产品了。因为我觉得这是很基础的能力，所以我一般都自己写，有时候想不到就把 `git diff` 的输出发给 ChatGPT，看它怎么写。

**Reply by hongtaoh** [reply]
因为有的更新很简单，比如修改一个 typo，我就懒得去弄。主要是我目前觉得没有必要，没有影响到我的效率，当我感觉我不总结不行的时候，就会去弄了。我的习惯是哪怕改一个 typo 我也马上 commit，而不是等之后。

**Reply by CyrusYip** [reply]
改错字可以直接写 fix typo。

以 https://github.com/hongtaoh/hugo-ht 这个开源项目为例。如果我是用户，看到提交信息（update on…）会觉得作者对这个项目不太上心，所以我大概率不会使用这个主题。如果我是开发者（想学习这个主题的代码）/贡献者，没有合理的提交信息将影响我看代码（我喜欢用 GitHub 的 blame 功能看某行代码的提交信息），于是放弃看这些代码，放弃贡献代码。

有提交信息也方便你自己回顾代码。有时候我写代码会用到一些 hack，这时写清楚注释和提交信息就很重要，因为这些代码看起来是很怪的。

不知道你是没考虑到这些，还是想着你的项目只是自己用？

**Reply by hongtaoh** [reply]
好问题。目前我还没有遇到过（1）我需要看我的 commit message (2) 别人需要看我的 commit message，所以我选择对我来说最有效率的办法。就目前我的所有项目来看，我感觉代码的注释和简单明了是最主要的。可能因为没有这方面需要，所以我从来都是用时间来写 commit message。

**Reply by hongtaoh** [reply]
慢慢在尝试着手写 git commit message: https://github.com/hongtaoh/kde_ebm
